### PR TITLE
Migrate to .NET 5

### PR DESCRIPTION
**Pull request by Dragemil**
Mostly just bumping up the numbers, however on resolving NRT warnings I kind of followed my guts, so please check if they are by any means sensible.

### CODE REVIEW

File: src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public async Task SendAsync(string message, string phoneNumber)
                 parameters["fast"] = "1";
             }
 
-            using (var body = new FormUrlEncodedContent(parameters))
+            using (var body = new FormUrlEncodedContent(parameters!))
```
Comment (marcin-dardzinski) [inline src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs:67]: Why `!` there? The object is clearly not null, does the compiler fail to infer that? :thinking:

**Review by lukaszgarstecki — APPROVED** [review]

File: Directory.Build.targets [hunk 2]
```diff
@@ -82,6 +85,8 @@
     <PackageReference Update="Microsoft.Extensions.Options.ConfigurationExtensions" Version="$(ExtensionsVersion)" />
     <PackageReference Update="Microsoft.Extensions.WebEncoders" Version="$(ExtensionsVersion)" />
 
+
+
```
Comment (Saancreed) [inline Directory.Build.targets]: Unnecessary newlines.

File: global.json [hunk 3]
```diff
@@ -1,6 +0,0 @@
-{
-  "sdk": {
-    "version": "3.1.100",
-    "rollForward": "latestMinor"
-  }
-}
```
Comment (Saancreed) [inline global.json]: Maybe we should keep it? Or is OmniSharp still refusing to cooperate?

File: src/Core/LeanCode.Components.Startup/IWebHostBuilderExtensions.cs [hunk 4]
```diff
@@ -28,27 +30,15 @@ public static IWebHostBuilder AddAppConfigurationFromAzureKeyVault(this IWebHost
...
-            Func<IConfiguration, AzureKeyVaultConfigurationOptions?> optionsProvider)
-        {
-            return builder.ConfigureAppConfiguration((context, builder) =>
-            {
-                var options = optionsProvider(builder.Build());
-
-                if (options != null)
-                {
-                    builder.AddAzureKeyVault(options);
+                    var vaultUlr = new Uri(vault);
```
Comment (Saancreed) [inline src/Core/LeanCode.Components.Startup/IWebHostBuilderExtensions.cs]: > vaultU**rl**

File: src/Core/LeanCode.Components.Startup/IWebHostBuilderExtensions.cs [hunk 5]
```diff
@@ -28,27 +30,15 @@ public static IWebHostBuilder AddAppConfigurationFromAzureKeyVault(this IWebHost
                 var configuration = builder.Build();
 
                 var vault = configuration.GetValue<string?>(VaultKey);
+                var tenantId = configuration.GetValue<string?>(TenantIdKey);
                 var clientId = configuration.GetValue<string?>(ClientIdKey);
                 var clientSecret = configuration.GetValue<string?>(ClientSecretKey);
 
-                if (vault != null && clientId != null && clientSecret != null)
+                if (tenantId != null && vault != null && clientId != null && clientSecret != null)
```
Comment (Saancreed) [inline src/Core/LeanCode.Components.Startup/IWebHostBuilderExtensions.cs]: Keep these checks in the same order as related variable declarations above.

File: src/Tools/LeanCode.ContractsGenerator/GeneratorConfiguration.cs [hunk 6]
```diff
@@ -26,7 +26,7 @@ public static async Task<List<GeneratorConfiguration>> GetConfigurations(string
             return await JsonSerializer.DeserializeAsync<List<GeneratorConfiguration>>(stream);
         }
 
-        public static List<GeneratorConfiguration> DeserializeConfigurationsFromString(string cfg)
+        public static List<GeneratorConfiguration>? DeserializeConfigurationsFromString(string cfg)
```
Comment (Saancreed) [inline src/Tools/LeanCode.ContractsGenerator/GeneratorConfiguration.cs]: I'd probably consider coercing `null`s to empty lists with `?? new List<GeneratorConfiguration> { }` instead.

**Review by Saancreed — CHANGES_REQUESTED** [review]

File: src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs (hunk 1 above)
Comment (Dragemil) [reply src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs:67]: _Argument of type 'Dictionary<string, string>' cannot be used for parameter 'nameValueCollection' of type 'IEnumerable<KeyValuePair<string?, string?>>' in 'FormUrlEncodedContent.FormUrlEncodedContent(IEnumerable<KeyValuePair<string?, string?>> nameValueCollection)' due to differences in the nullability of reference types._

File: global.json (hunk 3 above)
Comment (Dragemil) [reply global.json]: Works fine now, so I'm bringing this back

File: src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs (hunk 1 above)
Comment (marcin-dardzinski) [reply src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs:67]: Whoa, okay

File: Directory.Build.targets [hunk 7]
```diff
@@ -40,7 +44,7 @@
     <PackageReference Update="Google.Cloud.Firestore" Version="1.1.0" />
 
     <PackageReference Update="Microsoft.Azure.KeyVault" Version="3.0.5" />
-    <PackageReference Update="Microsoft.Data.SqlClient" Version="1.1.3" />
+    <PackageReference Update="Microsoft.Data.SqlClient" Version="2.0.0-preview3.20122.2" />
```
Comment (jakubfijalkowski) [inline Directory.Build.targets]: I _think_ 2.0.0 is newer than 2.0.0-preview3 :P

File: Directory.Build.targets [hunk 8]
```diff
@@ -1,8 +1,9 @@
 <Project>
   <PropertyGroup>
-    <EntityFrameworkCoreVersion>3.1.5</EntityFrameworkCoreVersion>
-    <AspNetCoreVersion>3.1.5</AspNetCoreVersion>
-    <ExtensionsVersion>3.1.5</ExtensionsVersion>
+    <EntityFrameworkCoreVersion>5.0.0-preview.6.*</EntityFrameworkCoreVersion>
+    <AspNetCoreVersion>5.0.0-preview.6.*</AspNetCoreVersion>
+    <ExtensionsVersion>5.0.0-preview.6.*</ExtensionsVersion>
+    <AzureVersion></AzureVersion>
```
Comment (jakubfijalkowski) [inline Directory.Build.targets]: What for?

File: src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs (hunk 1 above)
Comment (jakubfijalkowski) [reply src/Infrastructure/LeanCode.SmsSender/SmsApiClient.cs:67]: I would say - let's change line 45 to match the `FormUrlEncodedContent` signature.

**Review by jakubfijalkowski — CHANGES_REQUESTED** [review]

File: Directory.Build.targets (hunk 7 above)
Comment (Dragemil) [reply Directory.Build.targets]: **newer**?

Comment (jakubfijalkowski) [reply Directory.Build.targets]: newer, yeah, newer.

File: Directory.Build.targets (hunk 8 above)
Comment (Dragemil) [reply Directory.Build.targets]: Whoops, forgot to remove that

**Review by Saancreed — APPROVED** [review]

**Comment by Saancreed** [comment]
Don't forget to fix tests before merging :slightly_smiling_face:

File: .github/workflows/core_build_publish.yml [hunk 9]
```diff
@@ -16,7 +16,7 @@ jobs:
     - name: Setup .NET Core
       uses: actions/setup-dotnet@v1
       with:
-        dotnet-version: 3.1.202
+        dotnet-version: 5.0.100-preview.6.20318.15
```
Comment (jakubfijalkowski) [inline .github/workflows/core_build_publish.yml]: Let's bump to preview 7 now.

**Review by jakubfijalkowski — CHANGES_REQUESTED** [review]

**Comment by Dragemil** [comment]
This might be good to go now.

File: test/Domain/LeanCode.CQRS.RemoteHttp.Server.Tests/StubContext.cs [hunk 10]
```diff
@@ -30,7 +31,9 @@ public StubContext(string method, string path, string content)
         public override ConnectionInfo Connection => throw new NotImplementedException();
         public override WebSocketManager WebSockets => throw new NotImplementedException();
         public override ClaimsPrincipal User { get; set; }
+#pragma warning disable CS8609, CS8610 // It's absurd.
```
Comment (Saancreed) [inline test/Domain/LeanCode.CQRS.RemoteHttp.Server.Tests/StubContext.cs:34]: It's also probably wrong :smile:

**Review by Saancreed — APPROVED** [review]

**Review by jakubfijalkowski — APPROVED** [review]

**Pull request merged**